src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -23520,4 +23581,142 @@ .param [1]
+    {
+        var src = """
+42.M<object>();
+
+static class E
+{
+    extension<T>(T t) where T : struct
+    {
+        public void M() { System.Console.Write(t); }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (1,4): error CS0453: The type 'object' must be a non-nullable value type in order to use it as parameter 'T' in the generic type or method 'E.extension<T>(T)'
+            // 42.M<object>();
+            Diagnostic(ErrorCode.ERR_ValConstraintNotSatisfied, "M<object>").WithArguments("E.extension<T>(T)", "T", "object").WithLocation(1, 4));
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var invocation = GetSyntax<InvocationExpressionSyntax>(tree, "42.M<object>()");
+        Assert.Null(model.GetSymbolInfo(invocation).Symbol);
+    }
+
+    [Fact]
+    public void ExplicitTypeArguments_03()
+    {
+        var src = """
+string s = "ran";
+_ = s.P<object>;
+
+static class E
+{
+    extension<T>(T t)
+    {
+        public int P => 0;
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (2,7): error CS1061: 'string' does not contain a definition for 'P' and no accessible extension method 'P' accepting a first argument of type 'string' could be found (are you missing a using directive or an assembly reference?)
+            // _ = s.P<object>;
+            Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "P<object>").WithArguments("string", "P").WithLocation(2, 7));
+    }
+
+    [Fact]
+    public void ExplicitTypeArguments_04()
+    {
+        var src = """
+string s = "ran";
+s.M<string>();
+
+static class E
+{
+    extension<T>(T t)
+    {
+        public void M() => throw null;
+    }
+    extension(string s)
+    {
+        public void M<T>() { System.Console.Write("ran"); }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        CompileAndVerify(comp, expectedOutput: "ran").VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void ExplicitTypeArguments_05()
+    {
+        var src = """
+string s = null;
+s.M<string>();
+
+static class E
+{
+    extension<T>(T t)
+    {
+        public void M<U>() { }
+    }
+}
+""";
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (2,3): error CS1061: 'string' does not contain a definition for 'M' and no accessible extension method 'M' accepting a first argument of type 'string' could be found (are you missing a using directive or an assembly reference?)
+            // s.M<string>();
+            Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "M<string>").WithArguments("string", "M").WithLocation(2, 3));
+    }
+
+    [Fact]
+    public void ExplicitTypeArguments_06()
+    {
+        var src = """
+C<string, string>.M<string>();
+
+class C<T1, T2> { }
+static class E
+{
+    extension<T, U>(C<T, U> t)
+    {
+        public void M() { }
```

Comment:
Consider testing a similar case but where `M` is `static`.

Reply:
Second that. I think we need to add coverage for static members as well

Reply:
Thanks. That was a typo (intended for `static` here)